src/events/interactionCreate.mts

```diff
@@ -0,0 +1,34 @@
+// Finds and executes command when the slash command is used
+
+import { Events } from "discord.js";
+import { DiscordClient } from "../discordClient.mjs";
+
+export default {
+    name: Events.InteractionCreate,
+    once: false,
+    async execute(interaction) {
+        // Discard all non command usages (slash commands)
+        if (!interaction.isChatInputCommand()) return;
+
+        // Find command via commands
+        // We extended this from the base discord.js client to include a commands member
+        const command = (interaction.client as DiscordClient).commands.get(interaction.commandName);
```

Comment:
Try to avoid using `as T`, as it usually hides the fact that your variables aren't typed well.

Reply:
I _think_ in this case this is fine? iirc the only way to avoid doing using `as` is to also extend the interaction type so we indicate that we changed one member's type. However, it is your call here